openhtf/conf.py

```diff
@@ -46,6 +47,12 @@
     'This value will override any existing config value at config load time '
     'and will be a string')
 
+ConfigurationDeclaration = (
+    mutablerecords.Record(
+        'ConfigurationDeclaration',
+        ['name'],
+        {'description': None, 'default_value': None, 'optional': True}))
+
 
 class ConfigurationDeclarationError(Exception):
```

Comment:
It's not used elsewhere, I'll knock it out of this one.

That brings up something I found recently, voluptuous, another data-validation library (I can find them with ease now for some reason). It is really easy to specify schemas and it even supports this sort of mutually-exclusive set of keys in a dict. I'm considering adding support for it as a replacement for the plain list and dict here.